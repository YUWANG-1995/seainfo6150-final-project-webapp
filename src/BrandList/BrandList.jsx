import React from "react";
import { Link } from "react-router-dom";
import BrandListItem from "../BrandListItem/BrandListItem";
import styles from "./BrandList.module.css";

const BrandList = (props) => {
  return (
    <div className={styles.container}>
      <ul className={styles.carList}>
        {props.brands.map((brand, index) => (
        <Link key={index} to={"/cars/"+brand.brand}>
        <li key={index}>
            <BrandListItem brand={brand} key={index} />
        </li>
        </Link>
      ))}
      </ul>
    </div>
  );
}

export default BrandList;
