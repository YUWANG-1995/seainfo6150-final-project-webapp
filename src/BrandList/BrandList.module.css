.container{
  margin: 100px 70px;
}

.carList{
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  column-gap: 10px;
  list-style: none;
  overflow: hidden;
}

@media (min-width: 500px) {
  .carList{
    grid-template-columns: 1fr;
    column-gap: 30px;
  }
}

@media (min-width: 700px) {
  .carList{
    grid-template-columns: 1fr 1fr;
    column-gap: 25px;
  }
}

@media (min-width: 1100px) {
  .carList{
    grid-template-columns: repeat(3, 1fr);
    column-gap: 20px;
  }
}
