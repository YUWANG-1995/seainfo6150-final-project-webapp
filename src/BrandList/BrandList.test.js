import React from "react";
import { render } from "@testing-library/react";
import BrandList from "./BrandList.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the BrandList Component Correctly", () => {
  const test = [
    {
      brand: 'BMW',
      brandImage: {
        _url: "https://4.bp.blogspot.com/-IOuK59xvd-Q/V0yK19RcD8I/AAAAAAAAAJs/4gavimm0yW8av7K9yZSIvNQ5A-NeW2Y-ACLcB/s1600/bmw-logo.jpg"
      }
    },
    {
      brand: "Audi",
      brandImage: {
        _url: "https://www.logolynx.com/images/logolynx/3f/3f331726cd34688ab4080efc59e7017f.png"
      }
    }
  ]

  const { container } = render(
    <BrowserRouter>
      <BrandList
        brands={Object.values(test)}
      />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
