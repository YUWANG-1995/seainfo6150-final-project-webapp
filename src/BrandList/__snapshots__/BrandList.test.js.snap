// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the BrandList Component Correctly 1`] = `
<div>
  <div
    class="container"
  >
    <ul
      class="carList"
    >
      <a
        href="/cars/BMW"
      >
        <li>
          <div
            class="container"
          >
            <img
              alt="BMW"
              class="image"
              src="https://4.bp.blogspot.com/-IOuK59xvd-Q/V0yK19RcD8I/AAAAAAAAAJs/4gavimm0yW8av7K9yZSIvNQ5A-NeW2Y-ACLcB/s1600/bmw-logo.jpg"
            />
            <p
              class="brand"
            >
              BMW
            </p>
          </div>
        </li>
      </a>
      <a
        href="/cars/Audi"
      >
        <li>
          <div
            class="container"
          >
            <img
              alt="Audi"
              class="image"
              src="https://www.logolynx.com/images/logolynx/3f/3f331726cd34688ab4080efc59e7017f.png"
            />
            <p
              class="brand"
            >
              Audi
            </p>
          </div>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
